src/video.tsx

```diff
@@ -249,6 +258,37 @@ export class Video extends Component<VideoProps, VideoComponentState> {
     }
   };
 
+  private get isPictureInPictureActive(): boolean {
+    return !!getDocument()?.pictureInPictureElement;
+  }
+
+  private get isPictureInPictureEnabled(): boolean {
+    const { sourceType } = this.props;
+    return !!getDocument()?.pictureInPictureEnabled && sourceType === 'video';
+  }
+
+  private exitPictureInPicture = () => {
+    this.isPictureInPictureActive && getDocument()?.exitPictureInPicture();
+  };
+
+  private requestPictureInPicture = async () => {
+    if (!this.isPictureInPictureEnabled) {
+      return;
+    }
+    try {
+      // If the binary hasn't been loaded yet, this can throw an error
+      await this.videoRef.current?.requestPictureInPicture();
```

Comment:
Is it worth having a retry mechanism? aka, try again when the video finish loading

Reply:
I mean, not binary fully loaded, but the first bytes (i.e. no internet connection or similar).
The component should fall into an error state in these cases. Therefore, no need to auto retry here, I believe